extensions/shared/components/audio-player/index.jsx

```diff
@@ -0,0 +1,166 @@
+	onSkipForward,
+	onJumpBack,
+	currentTime,
+	playStatus = STATE_PAUSED,
+} ) {
+	const audioRef = useRef();
+
+	/**
+	 * Play current audio.
+	 *
+	 * @public
+	 */
+	const play = () => {
+		// Ignoring exceptions as they are handled globally from the audio element.
+		audioRef.current.play().catch( () => {} );
+	};
+
+	/**
+	 * Pause current audio.
+	 *
+	 * @public
+	 */
+	const pause = () => {
+		audioRef.current.pause();
+		speak( __( 'Paused', 'jetpack' ), 'assertive' );
+	};
+
+	useEffect( () => {
+		const audio = audioRef.current;
+		// Initialize MediaElement.js.
+		const mediaElement = new MediaElementPlayer( audio, meJsSettings );
+
+		// Add the skip and jump buttons if needed
+		if ( onJumpBack || onSkipForward ) {
+			const containerClass = `${ mediaElement.options.classPrefix }button ${ mediaElement.options.classPrefix }jump-button`;
+			if ( onSkipForward ) {
+				const buttonClass = `${ containerClass } ${ mediaElement.options.classPrefix }skip-forward-button`;
+				mediaElement.addControlElement(
+					createJumpButton( buttonClass, 'Skip Forward', onSkipForward ),
+					'skipForwardButton'
+				);
+			}
+
+			if ( onJumpBack ) {
+				const buttonClass = `${ containerClass } ${ mediaElement.options.classPrefix }jump-backward-button`;
+				mediaElement.addControlElement(
+					createJumpButton( buttonClass, 'Jump Back', onJumpBack ),
+					'jumpBackwardButton'
+				);
+			}
+		}
+
+		onPlay && audio.addEventListener( 'play', onPlay );
+		onPause && audio.addEventListener( 'pause', onPause );
+		onError && audio.addEventListener( 'error', onError );
+
+		return () => {
+			// Cleanup.
+			mediaElement.remove();
+			onPlay && audio.removeEventListener( 'play', onPlay );
+			onPause && audio.removeEventListener( 'pause', onPause );
+			onError && audio.removeEventListener( 'error', onError );
+		};
+	}, [ audioRef, onPlay, onPause, onError, onJumpBack, onSkipForward ] );
+
+	// If we get lots of events from clicking on the progress bar in the MediaElement
+	// then we can get stuck in a loop. We can so by debouncing here we wait until the
+	// next tick before acting on the playStatus prop value changing.
+	useEffect(
+		debounce( () => {
+			// Get the current status of the audio element and the required action to toggle it.
+			const [ audioStatus, action ] =
+				audioRef.current?.paused === false ? [ STATE_PLAYING, pause ] : [ STATE_PAUSED, play ];
+			if ( STATE_ERROR !== playStatus && audioStatus !== playStatus ) {
+				action();
+			}
+		} ),
+		[ audioRef, playStatus, trackSource ]
+	);
+
+	useEffect( () => {
+		if ( ! onTimeChange ) {
+			return;
+		}
+		//Add time change event listener
+		const audio = audioRef.current;
+		const throttledTimeChange = throttle( time => onTimeChange( time ), 1000 );
+		const onTimeUpdate = e => throttledTimeChange( e.target.currentTime );
+		onTimeChange && audio?.addEventListener( 'timeupdate', onTimeUpdate );
+
+		return () => {
+			audio?.removeEventListener( onTimeUpdate );
```

Comment:
When I insert the podcast player block from the inserter, I got the following error:

```
index.jsx:151 Uncaught TypeError: Failed to execute 'removeEventListener' on 'EventTarget': 2 arguments required, but only 1 present.
    at eval (index.jsx:151)
    at HTMLUnknownElement.callCallback (react-dom.82e849f1.js:182)
    at Object.invokeGuardedCallbackDev (react-dom.82e849f1.js:231)
    at invokeGuardedCallback (react-dom.82e849f1.js:286)
    at safelyCallDestroy (react-dom.82e849f1.js:19752)
    at react-dom.82e849f1.js:20225
    at unstable_runWithPriority (react.af754d50.js:2685)
    at runWithPriority$1 (react-dom.82e849f1.js:11174)
    at commitUnmount (react-dom.82e849f1.js:20218)
    at commitNestedUnmounts (react-dom.82e849f1.js:20298)
```

Reply:
I couldn't reproduce this, but it's definitely a bug. I've updated the code in 1a43b69